R/exclude.R

```diff
@@ -86,13 +97,22 @@ parse_exclusions <- function(file, exclude = settings$exclude,
   }
 
   start_locations <- rex::re_matches(lines, exclude_start, locations = TRUE)[, "end"] + 1L
+  end_locations <- rex::re_matches(lines, exclude_end, locations = TRUE)[, "start"]
   starts <- which(!is.na(start_locations))
-  ends <- which(rex::re_matches(lines, exclude_end))
+  ends <- which(!is.na(end_locations))
 
   if (length(starts) > 0) {
     if (length(starts) != length(ends)) {
-      starts_msg <- sprintf(ngettext(length(starts), "%d range start", "%d range starts"), length(starts))
-      ends_msg <- sprintf(ngettext(length(ends), "%d range end", "%d range ends"), length(ends))
+      starts_msg <- sprintf(
+        ngettext(length(starts), "%d range start (%s)", "%d range starts (%s)"),
```

Comment:
looking again, let's make this whole expression be the line_info helper function, just add an argument to toggle start vs end. lots of redundancy in the big sprintf() expression here

Reply:
good idea - done